File: Flowsource-templates/springboot3-azure-ci-pipeline-template/skeleton/src/main/java/${{values.java_package_name_slash}}/exception/RecordFailurePredicate.java
package ${{values.java_package_name}}.exception;

import java.util.function.Predicate;

/*
 * A custom Predicate which evaluates if an exception should be recorded as a failure. 
 * This predicate returns true for all exceptions other than Business Exception
 */
public class RecordFailurePredicate implements Predicate<Throwable> {
    @Override
    public boolean test(Throwable throwable) {
        return !(throwable instanceof BusinessException);
    }
}

File: Flowsource-templates/node-gcp-ci-template/skeleton/src/exception/RecordFailurePredicate.ts
import { HttpException } from '@nestjs/common';

export class BusinessException extends HttpException {
  constructor() {
    super('BusinessException', 400);
  }
}

export class RecordFailurePredicate {
  // A custom Predicate which evaluates if an exception should be recorded as a failure.
  // This predicate returns true for all exceptions other than Business Exception
  test(error: Error): boolean {
    return !(error instanceof BusinessException);
  }
}
